### ISSUE TITLE
Overwrite daemon flag from command line

### ISSUE THREAD

**Issue by ptejada** [APE-Project/APE_Server#40]
Here is a suggestion:

The below block is located in the in the file `ape.conf`:

```
Server {
        port = 6969
        daemon = no
        ip_listen = 0.0.0.0
        domain = auto
        rlimit_nofile = 10000
        pid_file = /var/run/aped.pid
}
```

It would help if this options could be overwritten through the command line

For example consider the below configuration:

```
Server {
        port = 6969
        daemon = yes
        ip_listen = 0.0.0.0
        domain = auto
        rlimit_nofile = 10000
        pid_file = ./running.pid
}
```

Executing `./aped` would start the server as daemon

To debug this server i have to duplicate the existing `ape.conf` into `debug.conf`. In `debug.conf` i just change `daemon = no` and then run `./aped --cfg debug.conf`

It would be nice if one could just run somthing like `./aped -daemon no` to debug a server which is configured to start as damon by default.

**Comment by verpeteren** [comment]
I just commited a solution, please test it on verpeteren/APE_Server 786928a6dad8e83e171e831558bfdcdb1916417a.

I am sure that there could be a better solution, but i guess this will work for now

Please test it, and also test it with http://ape-project.org/static/jsdocs/server/symbols/Ape.html#static_mainConfig 
as I need to store the 'overruled value' back in the 'config'.
The argc/argv is slightly different organised, adding other parameters would be a piece of cake.

**Comment by lcharette** [comment]
What is the difference between "Ape.config" and "Ape.mainConfig"? Shouldn't "Ape.config" returned the currently loaded config file? For other purpose, a third, optional parameter, could be added to "Ape.config" to specify which config file to used...?

**Comment by ptejada** [comment]
@lcharette `Ape.config` is used to load configurations files such as https://github.com/APE-Project/APE_Server/blob/master/modules/conf/inlinepush.conf

And `Ape.mainConfig` get the properties of ape.conf

**Comment by ptejada** [comment]
@verpeteren just gave it a quick test, and it works as expected. Running the following command:

```
./aped --daemon no
```

will run the server in the foreground, and so will any other string but _yes_:

```
./aped --daemon 123456
./aped --daemon ok
./aped --daemon yeah
```

**Comment by lcharette** [comment]
I have a question about this. Can you use `--deamon` and `--cfg` at the same time?

**Comment by verpeteren** [comment]
yes, you can use them simultaniously
e.g
./aped --daemon yes --cfg /etc/ape/ape.conf
./aped --daemon no  --cfg /etc/ape/ape.conf
./aped --daemon no
./aped --daemon yes
./aped  --cfg /etc/ape/ape.conf
./aped

**Comment by lcharette** [comment]
This is included in Pull Request #42. Closing this issue.
